docs/content/dagster-cloud/developing-testing/branch-deployments/using-branch-deployments-with-github.mdx

```diff
@@ -0,0 +1,302 @@
+   ```
+
+</details>
+
+---
+
+## Step 3: GitHub app
+
+TODO?
+
+---
+
+## Step 4: Configure the GitHub repository
+
+In this section, you'll:
+
+1. [Add the Docker registry to `cloud_workspace.yaml`](#step-41-add-the-docker-registry-to-cloud_workspaceyaml)
+2. [Configure GitHub Action secrets](#step-42-configure-github-action-secrets)
+3. [Update GitHub Workflow files](#step-43-update-github-workflow-files)
+4. [TODO: Verify builds]()
+
+### Step 4.1: Add the Docker registry to cloud_workspace.yaml
+
+In the `cloud_workspace.yaml` file, replace `build.registry` with the registry used by the [agent you created in Step 2](#step-2-create-and-configure-an-agent).
+
+For example:
+
+```yaml
+# cloud_workspace.yaml
+
+locations:
+  - location_name: example_location
+    code_source:
+      python_file: repo.py
+    build:
+      directory: ./example_location
+      registry: 764506304434.dkr.ecr.us-east-1.amazonaws.com/branch-deployment-agent
+```
+
+### Step 4.2: Configure GitHub Action secrets
+
+1. In the repository, click the **Settings** tab.
+2. In the **Security** section of the sidebar, click **Secrets > Actions**.
+3. Click **New repository secret**.
+4. In the **Name** field, enter the name of the secret. For example, `DAGSTER_CLOUD_API_TOKEN`
+5. In the **Value** field, paste the value of the secret.
+6. Click **Add secret**.
+
+Repeat steps 3-6 for each of the secrets required by your agent type:
+
+<details>
+  <summary><strong>AMAZON ECS AGENTS</strong></summary>
+
+- `DAGSTER_CLOUD_API_TOKEN` - The Dagster Cloud agent token you created in [Step 1](#step-1-generate-a-dagster-cloud-agent-token)
+- `ORGANIZATION_ID` - Your Dagster Cloud organization ID
+- `AWS_ACCESS_KEY` - The **Access key ID** of the AWS IAM user you created in [Step 2](#step-2-create-and-configure-an-agent)
+- `AWS_SECRET_ACCESS_KEY` - The **Secret access key** of the AWS IAM user you created in [Step 2](#step-2-create-and-configure-an-agent)
+
+The **Actions secrets** page should look like the following:
+
+<Image
+alt="Configured GitHub Actions secrets for an AWS ECS agent"
+src="/images/dagster-cloud/developing-testing/branch-deployments/github-aws-ecr.png"
+width={1091}
+height={623}
+/>
+
+</details>
+
+<details>
+  <summary><strong>DOCKER AGENTS</strong></summary>
+
+- `DAGSTER_CLOUD_API_TOKEN` - The Dagster Cloud agent token you created in [Step 1](#step-1-generate-a-dagster-cloud-agent-token)
+- `ORGANIZATION_ID` - Your Dagster Cloud organization ID
+- `DOCKERHUB_USERNAME` - Your DockerHub username
+- `DOCKERHUB_TOKEN` - <!-- https://docs.docker.com/docker-hub/access-tokens/#create-an-access-token -->
+
+The **Actions secrets** page should look like the following:
+
+<Image
+alt="Required GitHub Actions secrets for a Docker agent"
+src="/images/dagster-cloud/developing-testing/branch-deployments/github-dockerhub.png"
+width={790}
+height={318}
+/>
+
+</details>
+
+<details>
+  <summary>
+    <strong>KUBERNETES AGENTS</strong>
+  </summary>
+</details>
+
+<!--
```

Comment:
This was in the original BD guide - I'm not sure what to do with it.

- Do we support a Google version of an agent?
- Can GCR be used with another agent type?